glslc/src/file_compiler.cc

```diff
@@ -177,6 +177,24 @@ void FileCompiler::OutputMessages() {
   shaderc_util::OutputMessages(&std::cerr, total_warnings_, total_errors_);
 }
 
+namespace {
+std::string GetBaseNameFromAbsolutePath(const std::string& path) {
```

Comment:
Hmm. I see now that libshaderc_util/io.cpp has IsAbsolutePath, and that repeats some of the same logic here to determine which kind of path is being used.

So it would make sense to put this method into libshaderc_util.